src/utils/performance_monitor.py

```
@@ -0,0 +1,547 @@
+- Performance alerting and logging
+"""
+
+import logging
+import statistics
+import time
+from collections import defaultdict, deque
+from dataclasses import dataclass, field
+from datetime import datetime, timedelta
+from typing import Any
+
+import psutil
```

Pylint (3.3.7)

[error] Unable to import 'psutil'

(E0401)